### PR TITLE
feat: Add dark mode

### LINKED ISSUE #106: Add dark mode
From inspecting in Chrome / Nativefier:

Screen Shot 2021-05-12 at 9 26 41 am

Screen Shot 2021-05-12 at 9 26 44 am

### PR DESCRIPTION
**Pull request by minhhang107**
Resolves #106 
- added styles for dark/light theme
- added toggle component

### CODE REVIEW

**Comment by MichaelCurrin** [comment]
Thanks. Great stuff. I'll try it out.

File: src/components/ThemeToggler.vue [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        return "dark-theme";
+      } else {
+        return "light-theme";
+      }
+    },
+  },
+});
+</script>
+
+<!-- Add "scoped" attribute to limit CSS to this component only -->
```
Comment (MichaelCurrin) [inline src/components/ThemeToggler.vue:66]:
```suggestion
```

File: src/App.vue [hunk 2]
```diff
@@ -6,10 +6,13 @@
       <div id="nav">
         <!-- Build menu based on configured app routes. -->
         <!-- Keep router-link on one line for underlining -->
-        <span v-for="(item, index) in routes" :key="item.path">
+        <span class="nav-item" v-for="(item, index) in routes" :key="item.path">
           <span>{{ index !== 0 ? " | " : "" }}</span>
           <router-link :to="item.path">{{ item.name }}</router-link>
         </span>
+        <span class="nav-item">
+          <ThemeToggler></ThemeToggler>
```
Comment (MichaelCurrin) [inline src/App.vue:14]: I prefer using the short notation.
```suggestion
          <ThemeToggler/>
```

File: src/App.vue [hunk 3]
```diff
@@ -6,10 +6,13 @@
       <div id="nav">
         <!-- Build menu based on configured app routes. -->
         <!-- Keep router-link on one line for underlining -->
-        <span v-for="(item, index) in routes" :key="item.path">
+        <span class="nav-item" v-for="(item, index) in routes" :key="item.path">
```
Comment (MichaelCurrin) [inline src/App.vue:9]: This doesn't add any difference I can see so leave it out.
```suggestion
        <span v-for="(item, index) in routes" :key="item.path">
```

File: src/App.vue [hunk 4]
```diff
@@ -6,10 +6,13 @@
...
           <span>{{ index !== 0 ? " | " : "" }}</span>
           <router-link :to="item.path">{{ item.name }}</router-link>
         </span>
+        <span class="nav-item">
```
Comment (MichaelCurrin) [inline src/App.vue:13]: Can you rename the nav-item class?

```suggestion
        <span class="toggle">
```

File: src/App.vue [hunk 5]
```diff
@@ -75,7 +96,22 @@ export default defineComponent({
 
 #nav a.router-link-exact-active {
   text-decoration: underline;
-  color: var(--grey);
+  color: var(--accent-color);
+}
+
+.nav-item {
```
Comment (MichaelCurrin) [inline src/App.vue:102]:
```suggestion
.toggle {
```

File: src/components/ThemeToggler.vue [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+      ></div>
+    </label>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+
+export default defineComponent({
+  name: "ThemeToggler",
```
Comment (MichaelCurrin) [inline src/components/ThemeToggler.vue:24]: Can you rename this throughout as just `ThemeToggle` ?

File: src/components/ThemeToggler.vue [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+      localStorage.setItem("user-theme", theme);
+      this.userTheme = theme;
+      document.documentElement.className = theme;
+    },
+
+    getMediaPreference() {
+      const hasDarkPreference = window.matchMedia(
+        "(prefers-color-scheme: dark)"
+      ).matches;
+      if (hasDarkPreference) {
```
Comment (MichaelCurrin) [inline src/components/ThemeToggler.vue:56]:
```suggestion
      return hasDarkPreference ? "dark-theme" : "light-theme";
```

File: src/components/ThemeToggler.vue [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+  data() {
+    return {
+      userTheme: "light-theme",
+    };
+  },
+
+  methods: {
+    toggleTheme() {
+      const activeTheme = localStorage.getItem("user-theme");
+      if (activeTheme === "light-theme") {
```
Comment (MichaelCurrin) [inline src/components/ThemeToggler.vue:39]:
```suggestion
      const newTheme = activeTheme === "light-theme" ? "dark-theme" : "light-theme"
      this.setTheme(newTheme)
```

File: src/components/ThemeToggler.vue [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+        this.setTheme("dark-theme");
```
Comment (MichaelCurrin) [inline src/components/ThemeToggler.vue:40]:
```suggestion
```

File: src/components/ThemeToggler.vue [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+      } else {
```
Comment (MichaelCurrin) [inline src/components/ThemeToggler.vue:41]:
```suggestion
```

File: src/components/ThemeToggler.vue [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+        this.setTheme("light-theme");
```
Comment (MichaelCurrin) [inline src/components/ThemeToggler.vue:42]:
```suggestion
```

File: src/components/ThemeToggler.vue [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+      }
```
Comment (MichaelCurrin) [inline src/components/ThemeToggler.vue:43]:
```suggestion
```

**Review by MichaelCurrin — CHANGES_REQUESTED** [review]
Tried it out locally - all pages look good to me.

I suggested a refactor to simplify class use, refactor multiple lines into 1 or 2, and a rename of the toggler.

Thanks

File: src/components/ThemeToggler.vue (hunk 6 above)
Comment (minhhang107) [reply src/components/ThemeToggler.vue:24]: do you think I need to rename the file as ThemeToggle.vue as well?

Comment (MichaelCurrin) [reply src/components/ThemeToggler.vue:24]: yes - do a find and replace in the code and rename the file

```sh
$ git mv ThemeToggler.vue ThemeToggle.vue
$ git commit
```

**Comment by minhhang107** [comment]
Thanks for the suggestions. I've made modifications accordingly

**Comment by MichaelCurrin** [comment]
Brilliant. Appreciate it.

Merging. 🚀

**Review by MichaelCurrin — APPROVED** [review]

**Pull request merged**